### ISSUE TITLE
Is 1.10.3 usable?

### ISSUE THREAD

**Issue by siliconcat** [Azure/AKS#501]
Hi,

Yesterday I tried updating my 1.9.6 AKS to 1.10.3, but after doing that there seems to be a problem with certificates somewhere. Trying to `port-forward` or trying to display the dashboard will show problems such as:

```
error: error upgrading connection: unable to upgrade connection: Unauthorized
```

Tried to create a new 1.10.3 cluster and still getting the same issue. I can get the list of nodes, pods, and all that, but some actions, such as port-forwarding or trying to install a helm package, will lead to this message.

Does anyone know how to fix this?

In the meanwhile, will continue on 1.9.6...

**Comment by stigok** [comment]
I have success now with a 1.10.3 cluster. My 1.9.6 failed upgrade, so I created a new one starting from 1.10.3.

**Comment by siliconcat** [comment]
Did you enable RBAC? Maybe the issue is related to 1.10.3 with RBAC enabled?

**Comment by stigok** [comment]
No, I did not. I did not have time to debug why it failed, but it left my cluster in an unusable (Failed) state

**Comment by wojciech12** [comment]
I have upgraded 4 node cluster from 1.9.6 in west-europe. Works so far.

**Comment by siliconcat** [comment]
With RBAC enabled?

**Comment by wojciech12** [comment]
No, I just used the default upgrade. I thought RBACs are not supported by AKS. By mistake, the AKS team activated it on my 1.9.6 and things went south - https://github.com/Azure/AKS/issues/418#issuecomment-401889100

**Comment by siliconcat** [comment]
I see, the RBAC option is there in the ARM templates and not in the UI. For 1.9.6 it works fine, but as you say, it may not be properly supported yet and for 1.10.3 it just doesn't work... 

Thanks all for the insight!

**Comment by wojciech12** [comment]
The worst thing about AKS is how unpredictable it is. The AKS team seem to touch the k8s worker nodes a lot. Sometimes they introduce bugs and then you, a poor dev or ops, have no idea what has happen.

**Comment by angrox** [comment]
tunnelfront is still crashing randomly

**Comment by gorrog** [comment]
@angrox It's crashing for us too :(

**Comment by wojciech12** [comment]
@gorrog I had the same issue. A temporary solution was to delete all pods with kube-dns, kube-svc-redirect and tunnelfront. After they were recreated, it worked for few days. After that I have restarted one node after another. I do not have any issue since (my nodes: k8s 1.9.6, Ubuntu 16.04.4 LTS, Kernel: 4.13.0-1012-azure).

**Comment by wojciech12** [comment]
@gorrog @angrox again, DNS resolution stopped working in my cluster, next the tunnelfront... it was a dev cluster, so I just restarted node one by one. My nodes are on:
```
 v1.10.3   <none>        Ubuntu 16.04.4 LTS   4.15.0-1014-azure   docker://1.13.1
```

**Comment by carloreggiani** [comment]
I tryed to upgdare from 1.9.6 in west-europe a working cluster: failed.

Destroyed and recreated ... error in deploying :( 
I'm using the same name and the same VNET custom subnet, I'll trying to change some parameters. Not really an easy step.

**Comment by mjhoffman65** [comment]
tunnelfront is crashing for me today is west europe

**Comment by maetthu** [comment]
Same here. Freshly deployed 1.10.3 cluster (RBAC & advanced networking enabled), tunnelfront is crashing a lot:

```
$ kubectl get pods --all-namespaces
NAMESPACE     NAME                                          READY     STATUS    RESTARTS   AGE
[...]
kube-system   tunnelfront-658d448fcd-672sr                  1/1       Running   25         1d
```

**Comment by gorrog** [comment]
Update - the Azure team restarted the API layer on our cluster and this sorted it out. After upgrading we haven't had the issue happen again.
